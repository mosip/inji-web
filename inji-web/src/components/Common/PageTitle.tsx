import React from "react";

type PageTitleProps = {
    value: string;
    testId: string;
};

export function PageTitle({value, testId}: Readonly<PageTitleProps>) {
    return (
        <span
            data-testid={`title-${testId}`}
            className="text-2xl font-medium"
        >
            {value}
        </span>
    );
}