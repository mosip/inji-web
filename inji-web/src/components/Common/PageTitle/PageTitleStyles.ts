export const PageTitleStyles = {
    title: "text-2xl font-medium",
}