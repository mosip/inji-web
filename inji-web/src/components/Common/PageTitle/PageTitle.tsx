import React from "react";
import {PageTitleStyles} from "./PageTitleStyles";

type PageTitleProps = {
    value: string;
    testId: string;
};

export function PageTitle({value, testId}: Readonly<PageTitleProps>) {
    return (
        <span
            data-testid={`title-${testId}`}
            className={PageTitleStyles.title}
        >
            {value}
        </span>
    );
}