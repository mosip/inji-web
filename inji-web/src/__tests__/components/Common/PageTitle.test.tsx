import React from 'react';
import {render} from '@testing-library/react';
import {PageTitle} from '../../../components/Common/PageTitle/PageTitle';

describe('PageTitle Component', () => {
    it('should match snapshot', () => {
        const {asFragment} = render(
            <PageTitle value="Test Page Title" testId="test-page"/>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with correct value and test ID', () => {
        const {getByTestId} = render(
            <PageTitle value="Test Page Title" testId="test-page"/>
        );

        const titleElement = getByTestId('title-test-page');
        expect(titleElement).toBeInTheDocument();
        expect(titleElement).toHaveTextContent('Test Page Title');
        expect(titleElement).toHaveClass('text-2xl font-medium');
    });
});