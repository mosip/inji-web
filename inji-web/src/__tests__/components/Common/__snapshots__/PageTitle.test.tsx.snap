// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle Component should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="text-2xl font-medium"
    data-testid="title-test-page"
  >
    Test Page Title
  </span>
</DocumentFragment>
`;
